mautic/mautic
# Lead Bundle analysed by PHPSTAN

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | Y

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds the PHPSTAN analysis of LeadBundle to Travis CI. It also fixes all issues PHPSTAN found on level 0 analysis.

The obvious bug fix from the all changes is here: https://github.com/mautic/mautic/pull/4424/commits/a3b491798b3fad16e33eff0d49a30d943b5b1518 - I have no idea how Mautic could work with it. Calling a method which doesn't exist.

#### Steps to test this PR:
1. See Travis tests - should be green.
2. Click through the Lead functionalities in the UI. Everything should be same if not faster ;)

#### List deprecations along with the new alternative:
1. StagesChangeLogRepository::getMostStages()
2. StagesChangeLogRepository::getMostLeads()
3. StagesChangeLogRepository::countValue()
4. PointsChangeLogRepository::countValue()

Because I didn't find them to be used anywhere. It seems they were copied over from another class.
